consensus/leader.go

```diff
@@ -303,9 +303,15 @@ func (consensus *Consensus) onCommit(msg *msg_pb.Message) {
 			consensus.preCommitAndPropose(blockObj)
 		}
 
-		consensus.getLogger().Info().Msg("[OnCommit] Starting Grace Period")
 		go func(viewID uint64) {
-			time.Sleep(1000 * time.Millisecond)
+			waitTime := 1000 * time.Millisecond
+			maxWaitTime := time.Until(consensus.NextBlockDue)
```

Comment:
shouldn't you just recalculate the maxWaitTime based on the confirmed block timestamp? The consensus.NextBlockDue was updated during the block propose time, will it messed up with async block proposal? As it is the only variable, so there is chance that the nextblockdue was updated to the next next blockdue time.

Reply:
```				// Sleep to wait for the full block time
				consensus.getLogger().Info().Msg("[ConsensusMainLoop] Waiting for Block Time")
				**<-time.After(time.Until(consensus.NextBlockDue))**
				consensus.StartFinalityCount()

				// Update time due for next block
				consensus.NextBlockDue = time.Now().Add(consensus.BlockPeriod)```
it's updated after the last blockTime is passed. So no worry of timing issues.